earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -182,6 +313,12 @@ CreateEmptyDocument(const std::string &rootTagname) throw()
                                             0);// document type object (DTD)
     return doc;
    } catch (...) {
+     if (terminateCache)
+     {
+       xmlLibLock.Unlock();
```

Comment:
I'm not sure if this is safe here - you may be calling unlock without calling lock